### PR TITLE
chore: updating version.go adding more detail to Agent function

### LINKED ISSUE #823: Adding more details in agent hello and structured value
## Describe the problem you'd like to have solved

We have an **Agent** function in `version.go` that returns some details about the peer (for example `pactus/0.17.0`). We need a structured value and additional details added:

```go
func Agent() string {
	return fmt.Sprintf("pactus/%s", Version())
}
```

### Describe the ideal solution

We want agent details similar:

```go
func Agent() string {
	return fmt.Sprintf("node=pactus/version=%s/type={GUI or DAEMON}/os=%s/architecture=%s", Version(), runtime.GOOS, runtime.GOARCH)
}
```

In `fmt.Sprintf()` we have many **key=value** and split-ed by **/**

### PR DESCRIPTION
**Pull request by izikdepth**
- Fixes #823

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ const (
 )
 
 func Agent() string {
-	return fmt.Sprintf("pactus/%s", Version())
+	return fmt.Sprintf("node=pactus/version=%s/os=%s/architecture=%s", Version(), runtime.GOOS, runtime.GOARCH)
```
Comment (Ja7ad) [inline version/version.go:16]: We need another key=value about peer is from GUI or Daemon

**Review by Ja7ad — CHANGES_REQUESTED** [review]

File: version/version.go (hunk 1 above)
Comment (kehiy) [reply version/version.go:16]: @Ja7ad I checked the code, there are some places we are calling the Agent function but we don't have access to find out which client is calling the function.

should we pass it from other modules? (for example, adding a ClientType field to www modules, so in response we can use it to pass to the Aget)

**Comment by b00f** [comment]
@izikdepth @Ja7ad @kehiy

This PR is crucial, and we should ensure that we won't need to refactor it in the future. If you're all okay with it, let's schedule an internal meeting to discuss the best approach.


Question from @izikdepth 
Using above parameters: `runtime.GOOS`, `runtime.GOARCH`
Can we know if the agent is GUI or CLI?

**Comment by Ja7ad** [comment]
> @izikdepth @Ja7ad @kehiy
> 
> This PR is crucial, and we should ensure that we won't need to refactor it in the future. If you're all okay with it, let's schedule an internal meeting to discuss the best approach.
> 
> Question from @izikdepth Using above parameters: `runtime.GOOS`, `runtime.GOARCH` Can we know if the agent is GUI or CLI?

We can known with global variable or etc before init node, of course need this global has been set in `cmd`.

**Comment by kehiy** [comment]
> Question from @izikdepth 
> Using above parameters: `runtime.GOOS`, `runtime.GOARCH`
> Can we know if the agent is GUI or CLI?

We can have this meeting tomorrow or now if everyone ok(?).

**Comment by izikdepth** [comment]
tomorrow

> > Question from @izikdepth
> > Using above parameters: `runtime.GOOS`, `runtime.GOARCH`
> > Can we know if the agent is GUI or CLI?
> 
> We can have this meeting tomorrow or now if everyone ok(?).

tomorrow

**Comment by Ja7ad** [comment]
@izikdepth pip updated: https://pips.pactus.org/PIPs/pip-14

**Comment by kehiy** [comment]
Fixed in:
- #841

**Pull request closed without merging**